bndtools.m2e/src/bndtools/m2e/BndConfigurator.java

```diff
@@ -145,6 +145,9 @@ public AbstractBuildParticipant getBuildParticipant(final IMavenProjectFacade pr
 					return build;
 				}
 
+				final File classesDir = maven.getMojoParameterValue(projectFacade.getMavenProject(), execution,
+					"classesDir", File.class, monitor);
```

Comment:
What is `classesDir`? Are you reaching into BndMavenPlugin to read a field?

https://github.com/bndtools/bnd/blob/e5d1d948b5d323c5fefecf445c00feb3d0ba2e71/maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java#L95-L96

That seems highly fragile. And at the least deserves comments on both sides of the connection to document the connection. 

What if `includeClassesDir` is false?

Reply:
Ok, I've added a comment w.r.t. `classesDir`.

`includeClassesDir` won't have any effect on this change since it's not about the same thing. Even when `includeClassesDir` is false, `classesDir` is still the output dir we're getting bytecode and resources from.